### DISCUSSION TITLE
How to integrate into Electron?

### DISCUSSION THREAD

**Discussion by ghost** [noam-honig/radweb#225] [General]
Do you have any relevant cases or practices？

```code
# error
(node:66075) UnhandledPromiseRejectionWarning: Error: Task is not a known entity, did you forget to set @Entity() or did you forget to add the '@' before the call to Entity?
```

**Comment by noam-honig** [comment]
Hi @0x1016,

This error often comes in cases where an  `@` sign in missing before the Entity, or cases where more than one instance of remult is loaded into memory (esm & common JS or different versions of remult).

If you want to share an example project where you run into this problem, we'll be happy to help you find the cause in your specific code

**Reply by ghost** [reply]
hi @noam-honig , Really appreciate the response

This is a  [example project ](https://github.com/0x1016/electron-remult-test)
remult is a good project. This is my attempt in Electron.I have another question: How can I customize the port for calling APIs in remult? Currently, it conflicts with the Electron service Port. Thanks again.

**Reply by yoni-rapoport** [reply]
hi @0x1016,

I checked the example project and there seems to be an issue with electron-vite's `externalizeDepsPlugin` which doesn't treat the import from 'remult/remult-fastify' as an external dependency and bundles it into the 'out/main/index.js' file. A possible workaround is to manually add it as an external dependency in `electron.vite.config.ts`:

```
export default defineConfig({
  main: {
    plugins: [externalizeDepsPlugin()],
    build: {
      rollupOptions: {
        external: ['remult/remult-fastify']
      }
    }
  },
//....
```

As for telling Remult to use a different port for API requests - see https://remult.dev/docs/add-remult-to-your-app.html#changing-the-default-api-base-url.

(Note that this may cause CORS issues - no sure how that is handled with electron)

**Reply by ghost** [reply]
@yoni-rapoport  good job 🍻

**Reply by yoni-rapoport** [reply]
Looks like the next version of electron-vite will work without the workaround.

https://github.com/alex8088/electron-vite/pull/254
